pkg/scale/decode.go

```diff
@@ -555,27 +631,28 @@ var (
 )
 
 // decodeLength is helper method which calls decodeUint and casts to int
-func (ds *decodeState) decodeLength() (l uint, err error) {
+func (ds *decodeState) decodeLength() (l int, err error) {
```

Comment:
Why did we change the length from uint to int?

Reply:
Agreed, don't see the need to change this.

Reply:
Good question, I've changed this back.  (I'm not sure what my reasoning was for changing this).